src/error.rs

```diff
@@ -5,10 +5,17 @@ use std::fmt;
 /// The error returned when parsing a commit fails.
 #[derive(Clone, Debug, Eq, PartialEq)]
 pub struct Error {
-    commit: String,
-
     /// The kind of error.
     pub kind: Kind,
+
+    commit: Option<String>,
+}
+
+impl Error {
+    /// Create a new error from a `Kind`.
+    pub fn new(kind: Kind) -> Self {
```

Comment:
I've marked this `pub(crate)`.

I realise we already have a public `From`, but that's highly specific to `nom`, and I also consider that to be a mistake (I added it for internal convenience, without realising it would become part of the public API). I'll probably remove that impl before 1.0.

Reply:
My plan was to focus on improving `Error` in my next PR.